server/game/GameActions/AttachAction.ts

```diff
@@ -28,6 +35,8 @@ export class AttachAction extends CardGameAction {
             return false;
         } else if(!properties.attachment || properties.attachment.anotherUniqueInPlay(context.player) || !properties.attachment.canAttach(card, context)) {
             return false;
+        } else if(properties.takeControl && properties.attachment.controller === context.player) {
+            return false;
         }
         return card.allowAttachment(properties.attachment) && super.canAffect(card, context);
```

Comment:
Do we need a check that either the card is changing control or it's changing parent?